### PR TITLE
Add shortcut to a preheat menu

### LINKED ISSUE #20089: [FR] Direct access to preheat action
### Description

Direct preheat access from the temperature menu with material label and both temperatures shown. As seen on original prusa firmware. (no sub-menu or options for hotend and heated bed separately)
I tried adding this on my own but got lost on sub-menus and variables.

### Feature Workflow

1. Access temperature menu.
2. Select the preheat constant i want without accessing any sub-menus.
3. LCD goes back to status screen and preheating starts.

#### Additional Information

* You can see what i mean on this video (minute 11:10):
[Prusa video guide](https://youtu.be/GE-lrRbU124?t=670)

* The individual values for nozzle, bed and fans would be below the preheat constants on the same temperature menu as they are now, and the cooldown option at the bottom of the same menu.

* The preheat values would still be modified from the configuration menu.

* The preheat actual command should change whether the printer has heated bed and also adjust for multiple heaters/extruders, off course.

### PR DESCRIPTION
**Pull request by violetbp**
### Description
code for FR20089
Copy parts of menu_temperature to allow for easy addition of a 'shortcut' preheat menu in the main menu similar to prusa's firmware with Enable define in configuration.h
As the general need for preheating is both bed and hotend, removes the independent function in favor of having one less sub-menu.

edited:
- Configuration.h
- menu_temperature.cpp
- menu_main.cpp

### Benefits

Much easier to preheat a machine for those that may preform this action more than others

### Configurations
Any config file should work as long as it has the following define in it

```
// Insert a menu for preheating at the top level to allow for quick access 
#define LCD_PREHEAT_MENU
```

### Related Issues

https://github.com/MarlinFirmware/Marlin/issues/20089

### CODE REVIEW

**Comment by sjasonsmith** [comment]
@violetbp is this still behaving as you expect after the recent changes to this PR?

**Comment by rhapsodyv** [comment]
If we are starting to add shortcuts for user preferred actions, I think we should make it more generic. There're lots of different preferences and workflows, and this will open the doors for a lot of `FOO_SHORTCUT` being included in the configuration...

**Comment by thisiskeithb** [comment]
> If we are starting to add shortcuts for user preferred actions, I think we should make it more generic.

Indeed, but we have `CUSTOM_USER_MENUS` for that:

https://github.com/MarlinFirmware/Marlin/blob/e8ed880e6256704630009e1ae911bf476e09fd1a/Marlin/Configuration_adv.h#L3293-L3296

**Comment by rhapsodyv** [comment]
> CUSTOM_USER_MENUS

True! And the sample from the `Configuration_adv.h` is exactly for preheat!!

@violetbp is it enough for you?

File: Marlin/Configuration.h [hunk 1]
```diff
@@ -1556,6 +1556,9 @@
 #define PREHEAT_2_TEMP_BED    110
 #define PREHEAT_2_FAN_SPEED     0 // Value from 0 to 255
 
+// Insert a menu for preheating at the top level to allow for quick access
+#define PREHEAT_MENU_ITEM_SHORTCUT
```
Comment (rhapsodyv) [inline Marlin/Configuration.h]: Why not just use `CUSTOM_USER_MENUS`?

**Comment by vivianbp** [comment]
> > CUSTOM_USER_MENUS
> 
> True! And the sample from the `Configuration_adv.h` is exactly for preheat!!
> 
> @violetbp is it enough for you?

I'm not really the target audience for this feature, the idea is to make it super easy for anyone that doesn't want to get into custom menus and to not have to duplicate your heating settings from where they're specified.
edit: I should also point out that I know about custom menus, and was attempting to go off of what https://github.com/MarlinFirmware/Marlin/issues/20089 said

what do you think @wikolii

**Comment by wikolii** [comment]
1st of all, thanks everyone for their input.

I'm very familiar with the custom menus, i already use them. The thing is, the preheat constants you can use on those menus is static from what's in the firmware. If you change the constants on EEPROM you are stuck with the ones on the firmware if accessed from the custom menus.

I just thought  this feature was something more people would get behind.
Some 4 years ago i was able to do it myself, but then the code changed and i can't understand it in depth.

If anyone knows what to use on custom gcode to get the temp constants from eeprom that would also work, thanks!

File: Marlin/Configuration.h (hunk 1 above)
Comment (wikolii) [reply Marlin/Configuration.h]: Last i checked CUSTOM_USER_MENUS won't pick up the preheat constants saved on eeprom, only from firmware. You would have to upload the firmware again every time you want to change those shortcuts

Comment (rhapsodyv) [reply Marlin/Configuration.h]: This PR #20327 added a %t to get the ative extruder. I don't think it should be done here too, but that type of dynamic variable substitution would be very useful for `CUSTOM_USER_MENUS`.

Comment (wikolii) [reply Marlin/Configuration.h]: I'd be very happy with dynamic variables on CUSTOM_USER_MENUS.  Even better if i could see the dynamic value on the command label. e.g. "PLA **_210_** / **_45_**"

Comment (violetbp) [reply Marlin/Configuration.h]: should a FR be created for this?

**Comment by wikolii** [comment]
Hi again, will this be implemented anytime soon? 
Thanks!

**Comment by sjasonsmith** [comment]
This just needs @thinkyhead to make the decision of whether it is wanted or not.

**Comment by ETE-Design** [comment]
@wikolii Seems like you found a way to make custom menus other than the standard possible... Do you know how to make a "On / Off" Function, so it show the status depending on last command send... Like and relay On / Off so the Text show last status?

**Comment by thisiskeithb** [comment]
> Seems like you found a way to make custom menus other than the standard possible... Do you know how to make a "On / Off" Function, so it show the status depending on last command send... Like and relay On / Off so the Text show last status?

This Issue Queue is for Marlin bug reports and development-related issues, and we prefer not to handle user-support questions here. (As noted on [this page](https://github.com/MarlinFirmware/Marlin/blob/2.0.x/.github/contributing.md#i-dont-want-to-read-this-whole-thing-i-just-have-a-question).) For best results getting help with configuration and troubleshooting, please use the following resources:

- MarlinFW.org [Marlin Documentation](https://marlinfw.org)
- RepRap.org [Marlin Forum](https://forums.reprap.org/list.php?415)
- Facebook Group ["Marlin Firmware"](https://www.facebook.com/groups/1049718498464482/)
- Facebook Group ["Marlin Firmware for 3D Printers"](https://www.facebook.com/groups/3Dtechtalk/)
- [Marlin Configuration](https://www.youtube.com/results?search_query=marlin+configuration) on YouTube
- [Marlin Discord server](https://discord.gg/n5NJ59y)

**Pull request merged**